### PR TITLE
[#24927] Ensure the SDK harness handles JCL/log4j/log4j2 messages.

### LINKED ISSUE #24927: [Feature Request]: Ensure that JCL/log4j/log4j2 logging works from Java SDK harness
### What would you like to happen?

Currently JCL, log4j, and log4j2 messages are dropped in the Beam Java SDK harness.

### Issue Priority

Priority: 2 (default / most feature requests should be filed as P2)

### Issue Components

- [ ] Component: Python SDK
- [X] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [ ] Component: IO connector
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

### PR DESCRIPTION
**Pull request by lukecwik**
This routes JCL/log4j/log4j2 messages to SLF4 which is then routed to JUL and finally over the Fn Logging API to the runner.

Fixes #24927

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by lukecwik** [comment]
R: @apilloud @melaniedc

**Comment by lukecwik** [comment]
Java_PVR_Flink_Docker is one setup that covers the newly added test https://ci-beam.apache.org/job/beam_PreCommit_Java_PVR_Flink_Docker_Commit/2286/testReport/org.apache.beam.sdk/SdkHarnessEnvironmentTest/

File: sdks/java/io/elasticsearch-tests/elasticsearch-tests-common/build.gradle [hunk 1]
```diff
@@ -28,15 +28,6 @@ ext.summary = "Common test classes for ElasticsearchIO"
 def log4j_version = "2.17.1"
```
Comment (apilloud) [inline sdks/java/io/elasticsearch-tests/elasticsearch-tests-common/build.gradle:28]: nit: remove this?

File: buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy [hunk 2]
```diff
@@ -486,6 +486,7 @@ class BeamModulePlugin implements Plugin<Project> {
     def jsr305_version = "3.0.2"
     def everit_json_version = "1.14.1"
     def kafka_version = "2.4.1"
+    def log4j2_version = "2.19.0"
```
Comment (apilloud) [inline buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy:489]: Is there additional risk added by bumping the version? It was 2.17.1 before, not sure if this small version bump will be a problem. I think 2.17.1ish is in the Dataflow legacy worker container, will this conflict with that?

File: buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy [hunk 3]
```diff
@@ -573,6 +574,7 @@ class BeamModulePlugin implements Plugin<Project> {
         commons_csv                                 : "org.apache.commons:commons-csv:1.8",
         commons_io                                  : "commons-io:commons-io:2.7",
         commons_lang3                               : "org.apache.commons:commons-lang3:3.9",
+        commons_logging                             : "commons-logging:commons-logging:1.2",
```
Comment (apilloud) [inline buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy:577]: We explicitly remove commons_logging in flink with comments about jcl_over_slf4j. Do we need both?
https://github.com/apache/beam/blob/7d5d62e1ebf09eca0280bd7eea754a2223d0a416/runners/flink/job-server/flink_job_server.gradle#L75

**Review by apilloud — APPROVED** [review]
LGTM

File: buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy (hunk 3 above)
Comment (lukecwik) [reply buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy:577]: commons_logging is only used in the sdks/java/core tests, the flink override is about making sure we use slf4j

File: sdks/java/io/elasticsearch-tests/elasticsearch-tests-common/build.gradle (hunk 1 above)
Comment (lukecwik) [reply sdks/java/io/elasticsearch-tests/elasticsearch-tests-common/build.gradle:28]: done

File: buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy (hunk 2 above)
Comment (lukecwik) [reply buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy:489]: Dataflow legacy uses 2.17.2 which I swapped to here. Upgrading can happen in a separate PR.

**Comment by lukecwik** [comment]
Run Java_JDBC_IO_Direct PreCommit

**Comment by lukecwik** [comment]
Run Java PreCommit

**Comment by lukecwik** [comment]
Run Java_GCP_IO_Direct PreCommit

**Pull request merged**